### PR TITLE
[917] Clean up cookie logic

### PR DESCRIPTION
**Pull request by darokel**
### Context

- https://trello.com/c/hXNq3Ci0/1305-update-cookie-banner-logic-to-include-server-side-check

### Changes proposed in this pull request

- Refactor the cookie banner logic so it includes a server side check. We won't render the markup at all if the cookie has been set and avoid the "flickering" that is happening on the browser. If the user clears their cookies, the markup will display and allow the user to set again.
- Make the banner's cookie key and length configurable via `Settings`
- Refactor the cookie page routes, it was previously overriding the default Rails `cookies` method
- Simplify the cookie banner JS spec

### Guidance to review

- Check the cookie banner works as before

### CODE REVIEW

**Comment by darokel** [comment]
[[S] Update cookie banner logic to include server side check](https://trello.com/c/hXNq3Ci0/1305-update-cookie-banner-logic-to-include-server-side-check)

**Review by edwardhorsford — APPROVED** [review]
Nice improvement!

File: config/settings.yml [hunk 1]
```diff
@@ -50,6 +50,11 @@ sidekiq:
 pagination:
   records_per_page: 100
 
+cookies:
+  default_number_of_days: 182
+  cookie_banner_key: viewed_cookie_message
+
+
```
Comment (AbigailMcP) [inline config/settings.yml:57]: Extra line!

**Review by AbigailMcP — APPROVED** [review]
Looks good!

**Pull request merged**